OutOfSchool/OutOfSchool.WebApi/Services/Database/WorkshopService.cs

```diff
@@ -279,9 +279,13 @@ public async Task<SearchResult<WorkshopProviderViewCard>> GetByProviderId(Guid i
 
         var workshopProviderViewCards = mapper.Map<List<WorkshopProviderViewCard>>(workshops);
 
+        var workshopsIds = workshops.Select(x => x.Id).ToList();
+
         var query = chatrooms
             .SelectMany(room => room.ChatMessages, (room, message) => new { room, message })
-            .Where(entry => entry.message.ReadDateTime == null && !entry.message.SenderRoleIsProvider)
+            .Where(entry => entry.message.ReadDateTime == null
+                        && !entry.message.SenderRoleIsProvider
+                        && workshopsIds.Any(x => x == entry.room.WorkshopId))
```

Comment:
(Coding) Isn't `workshopsIds.Contains(entry.room.WorkshopId)` simpler?

Reply:
Yes, it is simpler. Thank you!